pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -51,11 +51,12 @@ def num_dataloaders(self) -> int:
     @property
     def dataloaders(self) -> Sequence[DataLoader]:
         """Returns the validation or test dataloaders."""
-        if self.trainer.testing:
-            return self.trainer.test_dataloaders
-        return self.trainer.val_dataloaders
+        dataloaders = self.trainer.test_dataloaders if self.trainer.testing else self.trainer.val_dataloaders
+        if dataloaders is None:
+            raise RuntimeError("Dataloaders should be available.")
```

Comment:
any case where this error will be raised here?

Reply:
This is just to please mypy, I could've used an assertion too, but since this is public and loops are public, I chose this option

Reply:
Should this be a ValueError ?

Reply:
yeah, I think assertion would be enough. That's what we have at other places.

Reply:
I prefer this one. It is cleaner. Ideally assertions only go into tests